src/constructs/loadbalancing/alb/application-load-balancer.ts

```diff
@@ -7,6 +7,17 @@ import type { GuMigratingResource } from "../../core/migrating";
 
 interface GuApplicationLoadBalancerProps extends ApplicationLoadBalancerProps, AppIdentity, GuMigratingResource {}
 
+/**
+ * Construct which creates an Application Load Balancer.
+ *
+ * This construct should be used in conjunction with [[`GuApplicationListener`]] and [[`GuApplicationTargetGroup`]]
```

Comment:
I know the pattern provides a way to wire these 3 constructs together, but writing these docs made me wonder if we should provide a single construct which bundles them together for the common case (e.g. a EC2 App which only accepts traffic over HTTPS).

Reply:
Yeah, that makes sense. It's tricky because I guess we have some layers of abstraction here:

1. Resource (CFN YAML)
2. CfnResource (CDK)
3. Construct (CDK)
4. GuConstruct (GuCDK)
5. GuPattern (GuCDK)

This feels like something in between 4 and 5, which is tricky but makes sense in this scenario. Our own GuConstructs can sometimes be a combination of multiple constructs, so I guess if we thought of the right naming/use-cases we could do it?